transactionparser/src/channeldetails.js

```diff
@@ -0,0 +1,38 @@
+const normalizeNodes = require('./utils').normalizeNodes;
+const dropsToXRP = require('./utils').dropsToXRP;
+const BigNumber = require('bignumber.js');
+
+function summarizePaymentChannel(node) {
+
+      const final = (node.diffType === 'CreatedNode') ? node.newFields : node.finalFields;
+      const prev = node.previousFields || {};
+
+      return {
+        channelId: node.ledgerIndex,
+        source: final.Account,
+        destination: final.Destination,
+        channel_amount_change: prev.Amount ? 
+          dropsToXRP(new BigNumber(final.Amount)).minus(dropsToXRP(new BigNumber(prev.Amount || 0))) : 
+          undefined,
+        channel_balance_change: final.Balance ? 
+          dropsToXRP(new BigNumber(final.Balance)).minus(dropsToXRP(new BigNumber(prev.Balance || 0))) : 
+          undefined,
+        channel_amount: 
+          dropsToXRP(new BigNumber(final.Amount || 0)),
+        channel_balance: 
+          dropsToXRP(new BigNumber(final.Balance || 0)),
+        prev_tx: node.PreviousTxnID
```

Comment:
This is missing some of the optional fields
https://developers.ripple.com/paychannel.html#paychannel-fields

Reply:
I think these are covered by the [existing payment channel parser](https://github.com/ripple/ripple-lib/blob/47a139fdff91b03062c8c401ccef365d876444d9/src/ledger/parse/payment-channel.ts#L23-L34)?